src/popup/components/MultisigVaultCreationProgress.vue

```diff
@@ -0,0 +1,199 @@
+      <div
+        v-for="step in VAULT_CREATION_STEPS_COUNT"
+        :key="step"
+        class="step-item"
+        :class="{
+          active: creatingVaultStep >= step,
+          completed: creatingVaultStep > step,
+        }"
+      >
+        <CheckSuccessCircleIcon
+          v-if="creatingVaultStep > step || !!multisigAccount"
+          class="step-success-icon"
+        />
+        <div
+          v-else
+          class="step-item-id"
+        >
+          {{ step }}
+        </div>
+
+        <div class="step-item-name">
+          {{ getStepContent(step) }}
+        </div>
+      </div>
+    </div>
+    <div
+      v-if="!!multisigAccount"
+      class="multisig-account-created"
+    >
+      <div class="message">
+        {{ $t('modals.creatingMultisigAccount.vaultCreatedMessage') }}
+      </div>
+      <AvatarWithChainName
+        :address="multisigAccount.gaAccountId"
+        class="ae-address"
+        show-address
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  PropType,
+  ref,
+} from '@vue/composition-api';
+import {
+  MULTISIG_CREATION_STEPS,
+} from '../utils';
+import { IMultisigAccount, IMultisigCreationStep } from '../../types';
+
+import AvatarWithChainName from './AvatarWithChainName.vue';
+import ProgressBar from './ProgressBar.vue';
+import PlusCircle from '../../icons/plus-circle-fill.svg?vue-component';
+import CheckSuccessCircleIcon from '../../icons/check-success-circle.svg?vue-component';
+
+export default defineComponent({
+  name: 'MultisigVaultCreationProgress',
+  components: {
+    ProgressBar,
+    AvatarWithChainName,
+    CheckSuccessCircleIcon,
+  },
+  props: {
+    multisigAccount: { type: Object as PropType<IMultisigAccount>, default: null },
+    progress: { type: String as PropType<IMultisigCreationStep>, required: true },
+  },
+  setup(props, { root }) {
+    const VAULT_CREATION_STEPS_COUNT = 4;
+    const creatingVaultStep = computed<number>(() => {
+      switch (props.progress) {
+        case MULTISIG_CREATION_STEPS.preparing:
+          return 1;
+        case MULTISIG_CREATION_STEPS.compiled:
+          return 2;
+        case MULTISIG_CREATION_STEPS.deployed:
+          return 3;
+        case MULTISIG_CREATION_STEPS.created:
+          return 4;
+        default:
+          return 1;
+      }
+    });
+    const stepProgress = 100 / VAULT_CREATION_STEPS_COUNT;
+    const isCreatingVault = ref<boolean>(false);
+    const creatingVaultProgress = computed<number>(() => (
+      stepProgress * creatingVaultStep.value
+    ));
+
+    function getStepContent(step: number) {
+      let progressStatus = 'pending';
+      if (creatingVaultStep.value === step) {
+        progressStatus = 'inProgress';
+      }
+
+      if (creatingVaultStep.value > step || !!props.multisigAccount) {
+        progressStatus = 'done';
+      }
+
+      return root.$t(`modals.creatingMultisigAccount.step${step}.${progressStatus}`);
```

Comment:
Please avoid parsing the translation paths as this makes much harder to establish if the translation string is actually used in the app.